src/BaseVault.sol

```diff
@@ -775,11 +781,44 @@ abstract contract BaseVault is IVault, ERC20PermitUpgradeable, AccessControlUpgr
     }
 
     function _processAccounting() internal virtual {
+        uint256 exchangeRateBefore = IERC4626(address(this)).convertToAssets(10 ** decimals());
+
         uint256 totalBaseBalance = computeTotalAssets();
 
         _getVaultStorage().totalAssets = totalBaseBalance;
         // solhint-disable-next-line not-rely-on-time
         emit ProcessAccounting(block.timestamp, totalBaseBalance);
+
+        address hooks_ = address(hooks());
+        if (hooks_ != address(0)) {
+            uint256 exchangeRateAfter = IERC4626(address(this)).convertToAssets(10 ** decimals());
+            uint256 totalSupplyBefore = totalSupply();
+            IHooks(hooks_).afterProcessAccounting(exchangeRateBefore, exchangeRateAfter, totalSupplyBefore);
```

Comment:
i'd pass in totalBaseAssetsBefore and totalBaseAssetsAfter

instead of a transform which is convertToAssets, so that the hook gets raw data

and processes accordingly, instead of reversing the computation if it needs to get totalAssets for whatever reason